packages/botonic-react/src/components/element.jsx

```diff
@@ -20,3 +20,41 @@ export const Element = props => {
   if (isBrowser()) return renderBrowser()
   else if (isNode()) return renderNode()
 }
+
+Element.serialize = elementProps => {
+  let element = Object.assign(
+    {},
+    ...elementProps.children
+      .filter(c => c && c.type && c.type.name != 'Button')
+      .map(c => c.type.serialize && c.type.serialize(c.props))
+  )
```

Comment:
This whole thing is confusing. The result of [...].filter.map() is an Array, then we're spreading it into an object?

Reply:
Yes, because we an element must be a single object, with their attributes. If an element is an array, when we try to decompose a carousel, we will obtain an array composed of multiple arrays ( one of only one length for every element ): Ex Carousel[ Element1 [{...atributes}], Element2 [{...at}], ...]